src/unistd.rs

```diff
@@ -756,6 +756,36 @@ pub fn gethostname<'a>(buffer: &'a mut [u8]) -> Result<&'a CStr> {
     })
 }
 
+/// Set the NIS domain name (see
+/// [setdomainname(2)](http://man7.org/linux/man-pages/man2/setdomainname.2.html)).
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub fn setdomainname<S: AsRef<OsStr>>(name: S) -> Result<()> {
+    let ptr = name.as_ref().as_bytes().as_ptr() as *const c_char;
+    let len = name.as_ref().len() as size_t;
+
+    let res = unsafe { libc::setdomainname(ptr, len) };
+    Errno::result(res).map(drop)
+}
+
+/// Get the domain name and store it in the provided buffer, returning a pointer
+/// the CStr in that buffer on success (see
+/// [getdomainname(2)](http://man7.org/linux/man-pages/man2/getdomainname.2.html)).
+///
+/// This function call attempts to get the NIS domain name for the running system
+/// and store it in a provided buffer.  The buffer will be populated with bytes up
+/// to the length of the provided slice including a NUL terminating byte.
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub fn getdomainname<'a>(buffer: &'a mut [u8]) -> Result<&'a CStr> {
```

Comment:
I'm not super-happy about the ergonomics of this signature, but I just followed `gethostname` path. Can we rework these to allocate internally and return an owning `CString`?